codegen/src/generator/rest_xml.rs

```diff
@@ -0,0 +1,855 @@
+            }}",
+            field_name = generate_field_name(member_name),
+            parse_expression = parse_expression,
+            location_name = location_name,
+        ))
+
+    }).collect::<Vec<String>>().join("\n")
+}
+
+fn generate_struct_field_parse_expression(
+    shape: &Shape,
+    member_name: &str,
+    member: &Member,
+    location_name: Option<&String>,
+) -> String {
+
+    let location_to_use = match location_name {
+        Some(loc) => loc.to_string(),
+        None => member_name.to_string(),
+    };
+    let expression = format!(
+        "try!({name}Deserializer::deserialize(\"{location}\", stack))",
+        name = member.shape,
+        location = location_to_use,
+    );
+
+    if shape.required(member_name) {
+        expression
+    } else {
+        format!("Some({})", expression)
+    }
+}
+
+fn generate_serializer_body(shape: &Shape, service: &Service) -> String {
+    match shape.shape_type {
+        ShapeType::List => generate_list_serializer(shape),
+        ShapeType::Map => generate_map_serializer(shape),
+        ShapeType::Structure => generate_struct_serializer(shape, service),
+        _ => generate_primitive_serializer(shape),
+    }
+}
+
+fn generate_serializer_signature(name: &str) -> String {
+    format!("
+        #[allow(unused_variables, warnings)]
+        pub fn serialize(name: &str, obj: &{}) -> String", name)
+}
+
+fn generate_primitive_serializer(shape: &Shape) -> String {
+     let value_str = match shape.shape_type {
+        ShapeType::Blob => "String::from_utf8(obj.to_vec()).expect(\"Not a UTF-8 string\")",
+        _ => "obj.to_string()"
+     };
+     format!("format!(\"<{{name}}>{{value}}</{{name}}>\", 
+                name = name, 
+                value = {value_str})",
+                value_str = value_str)
+
+}
+
+fn generate_list_serializer(shape: &Shape) -> String {
+
+    // flattened lists don't have enclosing <FooList> tags
+    // around the list elements
+    let flattened = match shape.flattened {
+        Some(true) => true,
+        _ => false
+    };
+
+    let member = shape.member.as_ref().unwrap();
+    let element_type = &generate_member_name(&shape.member()[..]);
+    let location_name = match member.location_name {
+        Some(ref name) => name,
+        None => element_type
+    };
+
+    let mut serializer = format!("let mut parts: Vec<String> = Vec::new();");
+
+    if !flattened {
+        serializer += &format!("parts.push(format!(\"<{{}}>\", name));");
+    }
+
+    serializer += &format!("
+        for element in obj {{
+            parts.push({element_type}Serializer::serialize(\"{location_name}\", element));
+        }}
+        ",
+        element_type = element_type,
+        location_name = location_name);
+
+    if !flattened {        
+        serializer += &format!("parts.push(format!(\"</{{}}>\", name));");
+    }
+
+
+    serializer += "parts.join(\"\")";
+    serializer
+
+}
+
+fn generate_map_serializer(_shape: &Shape) -> String {
```

Comment:
Actually, that function is completely unimplemented.  Good catch.  This means that S3 doesn't need to serialize Map types in request bodies (or at least not for any of the methods I've tested), but I'd be willing to bet that this will be needed by _something_ with a rest-xml protocol.  I'll implement that before this gets merged.